### ISSUE TITLE
Use `#' @examplesIf` roxygen directive over `if()` in the docs

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#318]
There are a couple of reasons to do this:

1. When there are more than two outputs in `if()`, only the final one will be displayed in the _pkgdown_ website (cf. https://github.com/r-lib/pkgdown/issues/2175). 

```r
#' @examples
#' if (getRversion() > "4.0") {
#'   sd(cars$speed)
#'   sd(cars$dist)
#' }
```

<img width="769" alt="Screenshot 2022-10-21 at 21 13 07" src="https://user-images.githubusercontent.com/11330453/197272067-36a6e371-d7e9-4d2d-88f8-737379ac7843.png">

This is not the case if we use  `#' @examplesIf` instead.

```r
#' @examplesIf getRversion() > "4.0"
#' sd(cars$speed)
#' sd(cars$dist)
```

<img width="614" alt="Screenshot 2022-10-21 at 21 12 50" src="https://user-images.githubusercontent.com/11330453/197272188-4c437222-1d62-43b7-b16d-a95de8f58024.png">

2. This also helps clarify all external dependencies at the top-level (which helps with #317).

# Progress tracker:

- [x] insight 
- [ ] modelbased
- [ ] bayestestR
- [x] effectsize
- [ ] parameters
- [ ] performance
- [ ] correlation
- [x] report
- [x] see
- [x] datawizard
- [x] easystats

**Comment by IndrajeetPatil** [comment]
Keep in mind that if your original conditional statement was something like:

```r
#' @examples 
if (requireNamespace("zoo", quietly = TRUE)) {
  ...
}
```

If you use `examplesIf`, this won't be included in the output, so it might be a good idea to either namespace all externally used functions, or have a `library()` call at the top so that the users can exactly reproduce the output.

```r
#' @examplesIf requireNamespace("zoo", quietly = TRUE) 
library(zoo)
 ...
```

Thanks, @etiennebacher, for bringing up this issue.

**Comment by mattansb** [comment]
Note that in R 4.2.1, the "run examples" is still meh here...

![image](https://user-images.githubusercontent.com/35330040/197277436-55cd3f90-0f8a-4384-a1b6-999ee98bac77.png)

**Comment by IndrajeetPatil** [comment]
Hmm, but this is not too bad. At least, it shows all the output. I am a bit more concerned about some of the outputs not showing up at all on the website.

---

I can create an issue in rstudio repo to check if they can handle this a bit better. Ideally, the following should not appear in the output:

```r
## Don't show: 
## End(Don't show)
}) # examplesIf
```

**Comment by IndrajeetPatil** [comment]
I also checked in the PDF manual, and the examples look good.

**Comment by IndrajeetPatil** [comment]
Created issue in RStudio to track the suboptimal rendering of examples when run in the IDE: https://github.com/rstudio/rstudio/issues/12318

**Comment by IndrajeetPatil** [comment]
@rempsyc It would be great if you can also resolve this for _report_ before the next submission.

**Comment by rempsyc** [comment]
Is this necessary when the example is already wrapped in `\donttest`?

**Comment by IndrajeetPatil** [comment]
Yes, because CRAN tests `\donttest` examples in [an additional check](https://www.stats.ox.ac.uk/pub/bdr/donttest/).

**Comment by IndrajeetPatil** [comment]
_bump_

**Comment by IndrajeetPatil** [comment]
*bump*

**Comment by strengejacke** [comment]
> Keep in mind that if your original conditional statement was something like:
> 
> ```r
> #' @examples 
> if (requireNamespace("zoo", quietly = TRUE)) {
>   ...
> }
> ```
> 
> If you use `examplesIf`, this won't be included in the output, so it might be a good idea to either namespace all externally used functions, or have a `library()` call at the top so that the users can exactly reproduce the output.
> 
> ```r
> #' @examplesIf requireNamespace("zoo", quietly = TRUE) 
> library(zoo)
>  ...
> ```
> 
> Thanks, @etiennebacher, for bringing up this issue.

What about `@ExamplesIf require(<package>)` ?

**Comment by IndrajeetPatil** [comment]
Yepp, this is fine. In fact, it's one of the very few situations where `require()` usage is acceptable.

**Comment by bwiernik** [comment]
Could you explain the difference here? Not sure I'm following

**Comment by IndrajeetPatil** [comment]
https://r-pkgs.org/dependencies-mindset-background.html#sec-dependencies-attach-vs-load

<img width="773" alt="Screenshot 2024-06-05 at 21 34 55" src="https://github.com/easystats/easystats/assets/11330453/252729f4-2be4-40c0-b2f8-a097baedf4c6">

**Comment by strengejacke** [comment]
I think we have implemented this in most/nearly all instances...
